scripts/azure/util.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+#
+# Define functions to customize the Kubeflow app for Azure.
+#
+set -xe
+
+validate_az_arg() {
+    if [[ -z "$AZ_CLIENT_ID" ]]; then
+      echo "Azure service principal client id must be provided using --azClientId <AZ_CLIENT_ID>"
+      exit 1
+    fi
+    if [[ -z "$AZ_CLIENT_SECRET" ]]; then
+      echo "Azure service principal client secret must be provided using --azClientSecret <AZ_CLIENT_SECRET>"
+      exit 1
+    fi
+    if [[ -z "$AZ_TENANT_ID" ]]; then
+      echo "Azure tenant id must be provided using --azTenantId <AZ_TENANT_ID>"
+      exit 1
+    fi
+    if [[ -z "$AZ_SUBSCRIPTION_ID" ]]; then
+      echo "Azure subscription id must be provided using --azSubscriptionId <AZ_SUBSCRIPTION_ID>"
+      exit 1
+    fi
+    if [[ -z "$AZ_LOCATION" ]]; then
+      echo "Azure location must be provided using --azLocation <AZ_LOCATION>"
+      exit 1
+    fi
+}
+
+check_az_cli() {
+
+  if ! which "az" &>/dev/null && ! type -a "az" &>/dev/null ; then
+    echo "You don't have az cli installed. Please install az cli. https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-linux?view=azure-cli-latest"
+    exit 1
+  fi
+}
+
+az_login() {
+    if ! az login --service-principal --username "${AZ_CLIENT_ID}" --password "${AZ_CLIENT_SECRET}" --tenant "${AZ_TENANT_ID}" &>/dev/null ; then
+      echo "az login failed. Please check the env variables provided, then try again."
+      exit 1
+    fi
+    if ! az account set -s $AZ_SUBSCRIPTION_ID &>/dev/null ; then
+      echo "az account set -s ${AZ_SUBSCRIPTION_ID} failed."
+      exit 1
+    fi
+}
+
+createAKSCluster() {
+    if [ `az group exists --name "${DEPLOYMENT_NAME}"` == "false" ] ; then
+        if ! az group create --name "${DEPLOYMENT_NAME}" --location "${AZ_LOCATION}" &>/dev/null ; then
+            echo "az group create failed."
+            exit 1
+        fi
+    else
+        echo "Resource group ${DEPLOYMENT_NAME} already exists, skip az group create."
+    fi
+    if ! az aks get-credentials --resource-group=${DEPLOYMENT_NAME} --name=${DEPLOYMENT_NAME} 2>/dev/null ;
+    then
+        echo "Kubernetes cluster not found, creating."
+        az aks create --resource-group ${DEPLOYMENT_NAME} --name=${DEPLOYMENT_NAME} --node-count 3 --node-vm-size "Standard_NC6" --kubernetes-version 1.12.5 --generate-ssh-keys -l "${AZ_LOCATION}" --service-principal ${AZ_CLIENT_ID} --client-secret "${AZ_CLIENT_SECRET}" 
```

Comment:
should the sku be customizable? this is a pretty old sku

Reply:
done